### ISSUE TITLE
OpenAIChat skill fails to build

### ISSUE THREAD

**Issue by rodrigobdz** [FurhatRobotics/example-skills#41]
Running the [main.kt](https://github.com/FurhatRobotics/example-skills/blob/7ff51c1a08593f78ddb8422009ad077df35329e4/OpenAIChat/src/main/kotlin/furhatos/app/openaichat/main.kt), as suggested in the [docs](https://docs.furhat.io/skills/#running-a-skill-from-directly-from-intellij-with-the-virtual-furhat), fails with the following error:

```
e: .../example-skills/OpenAIChat/src/main/kotlin/furhatos/app/openaichat/flow/chatbot/openai.kt: (13, 19): Unresolved reference: OpenAiService
```

A [search](https://github.com/search?q=repo%3ATheoKanning%2Fopenai-java+com.theokanning.openai.OpenAiService&type=code) for [`com.theokanning.openai.OpenAiService`](https://github.com/FurhatRobotics/example-skills/blob/7ff51c1a08593f78ddb8422009ad077df35329e4/OpenAIChat/src/main/kotlin/furhatos/app/openaichat/flow/chatbot/openai.kt#L3) didn't return any results on the [third-party library](https://github.com/TheoKanning/openai-java).

The only similar import I could find was [`com.theokanning.openai.service.OpenAiService;`](https://github.com/TheoKanning/openai-java/blob/3a5bc281662a895553583b8d3dd9ce8112035996/example/src/main/java/example/OpenAiApiExample.java#L6).

**Comment by SquaredCanine** [comment]
Thanks for bringing this to our attention. The import you found is the correct one! We'll make sure to fix it :)

**Comment by rodrigobdz** [comment]
@SquaredCanine Thank you for your prompt reply. I tried using the new library (changes in diff below) but the issue persists (see screenshot).

Am I missing something?

---

```diff
- com.theokanning.openai.OpenAiService
+ com.theokanning.openai.service.OpenAiService
```

<img width="2032" alt="image" src="https://github.com/FurhatRobotics/example-skills/assets/14152377/ade517bf-2188-4b79-a371-7f415b5f915e">

**Comment by rodrigobdz** [comment]
Probably the following additional dependency is required in [build.gradle](https://github.com/FurhatRobotics/example-skills/blob/7ff51c1a08593f78ddb8422009ad077df35329e4/OpenAIChat/build.gradle#L47-L50):

```diff
dependencies {
    compile 'com.furhatrobotics.furhatos:furhat-commons:+'
    implementation 'com.theokanning.openai-gpt3-java:client:+'
+   implementation 'com.theokanning.openai-gpt3-java:service:+'
}
```

Trying this out now.

**Comment by rodrigobdz** [comment]
It still wasn't working, so I proceeded to clear the Gradle caches in the hope to ensure that the newly added library would be downloaded.

```sh
rm -rf $HOME/.gradle/caches/
```

Afterwards, I started getting the following error when trying to build `main.kt`. The build is broken 😰.

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/schettino/Downloads/example-skills/OpenAIChat/build.gradle' line: 56

* What went wrong:
A problem occurred evaluating root project 'OpenAIChat'.
> Could not resolve all files for configuration ':compile'.
   > Could not resolve khttp:khttp:1.0.0.
     Required by:
         project : > com.furhatrobotics.furhatos:furhat-commons:2.6.1
      > Could not resolve khttp:khttp:1.0.0.
         > Could not get resource 'https://s3-eu-west-1.amazonaws.com/furhat-maven/releases/khttp/khttp/1.0.0/khttp-1.0.0.pom'.
            > Could not GET 'https://s3-eu-west-1.amazonaws.com/furhat-maven/releases/khttp/khttp/1.0.0/khttp-1.0.0.pom'. Received status code 403 from server: Forbidden

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
```

I saw this related issue https://github.com/FurhatRobotics/tutorials/issues/9, but the proposed solution is already implemented. Even twice, seems like a typo. Anyways, any ideas how to fix the build?

https://github.com/FurhatRobotics/example-skills/blob/7ff51c1a08593f78ddb8422009ad077df35329e4/OpenAIChat/build.gradle#L40-L43

**Comment by MathieuRH** [comment]
Hey ! Could you try this line for the dependencies ? 
`implementation 'com.theokanning.openai-gpt3-java:client:0.12.0'`
The Import brought breaking changes starting 0.13.0
Hope that this fixes your problem !

**Comment by rodrigobdz** [comment]
@MathieuRH Thanks, but I am getting the following error:

> Could not GET 'https://s3-eu-west-1.amazonaws.com/furhat-maven/releases/khttp/khttp/1.0.0/khttp-1.0.0.pom'. Received status code 403 from server: Forbidden

It is failing during the build, before the proposed changes are loaded.

**Comment by rodrigobdz** [comment]
I found related [documentation](https://docs.furhat.io/bintray_sundown/#old-build-gradle) about the deprecation of `jcenter()`. Please help me get the skill up and running.

**Comment by MathieuRH** [comment]
Fixed in the #42 merge request
